examples/camera-app/linux/src/clusters/push_av_transport/push-av-transport-manager.cpp

```diff
@@ -0,0 +1,209 @@
+        return Status::NotFound;
+    }
+
+    mTransportOptionsMap[connectionID] = transportOptions;
+
+    ChipLogProgress(Camera, "PushAvStreamTransportManager, failed to find Connection :[%u]", connectionID);
+
+    return Status::Success;
+}
+
+Protocols::InteractionModel::Status PushAvStreamTransportManager::SetTransportStatus(const std::vector<uint16_t> connectionIDList,
+                                                                                     TransportStatusEnum transportStatus)
+{
+    if (connectionIDList.empty())
+    {
+        ChipLogError(Camera, "PushAvStreamTransportManager, connectionIDList is empty");
+        return Status::Failure;
+    }
+
+    if (transportStatus == TransportStatusEnum::kUnknownEnumValue)
+    {
+        ChipLogError(Camera, "PushAvStreamTransportManager, Invalid TransportStatus, transportStatus: [%u]",
+                     (uint16_t) transportStatus);
+        return Status::Failure;
+    }
+
+    for (uint16_t connectionID : connectionIDList)
+    {
+        if (mTransportMap.find(connectionID) == mTransportMap.end())
+        {
+            ChipLogError(Camera, "PushAvStreamTransportManager, failed to find Connection :[%u]", connectionID);
+            continue;
+        }
+        mTransportMap[connectionID]->setTransportStatus(transportStatus);
+    }
+
+    return Status::Success;
+}
+
+Protocols::InteractionModel::Status PushAvStreamTransportManager::ManuallyTriggerTransport(
+    const uint16_t connectionID, TriggerActivationReasonEnum activationReason,
+    const Optional<Structs::TransportMotionTriggerTimeControlStruct::DecodableType> & timeControl)
+{
+    if (activationReason == TriggerActivationReasonEnum::kUnknownEnumValue)
+    {
+        ChipLogError(Camera, "PushAvStreamTransportManager, Manual Trigger failed for connection [%u], reason: [%u]", connectionID,
+                     (uint16_t) activationReason);
+        return Status::Failure;
+    }
+
+    if (mTransportMap.find(connectionID) == mTransportMap.end())
+    {
+        ChipLogError(Camera, "PushAvStreamTransportManager, failed to find Connection :[%u]", connectionID);
+        return Status::NotFound;
+    }
+
+    ChipLogProgress(Camera, "PushAvStreamTransportManager, Trigger PushAV Transport for Connection: [%u]", connectionID);
+    mTransportMap[connectionID]->TriggerTransport(activationReason);
+
+    return Status::Success;
+}
+
+Protocols::InteractionModel::Status
+PushAvStreamTransportManager::FindTransport(const Optional<DataModel::Nullable<uint16_t>> & connectionID,
+                                            DataModel::List<const TransportConfigurationStruct> & outtransportConfigurations)
+{
+    if (connectionID.Value().IsNull())
+    {
+        ChipLogError(Camera, "PushAvStreamTransportManager, connectionID is null");
+        return Status::Failure;
+    }
+
+    std::vector<TransportConfigurationStruct> configList;
```

Comment:
You build a local std::vector<…> configList, then wrap it in a DataModel::List that stores raw pointers into that vector. Once the function returns the vector is destroyed ➜ dangling pointers.